### ISSUE TITLE
Mod_muc destroys rooms on node exit.

### ISSUE THREAD

**Issue by rmelcer** [processone/ejabberd#342]
Hi all,

(I'm not sure if this would be considered a feature-request or a bug. I consider it a bug, but the current behavior could be intentional.)

Even after spending quite a bit of time looking at this, I still find erlang mystifying, so I don't have a patch included for you. Sorry, I know that can be a deal-breaker in some places.

I have two ejabberd nodes in a cluster. Everything works perfectly, as far as I can tell. The problem I'm having is that when a node gets shut down normally, mod_muc destroys all of the rooms that exist on that node.

So if my nodes are node1.example.com and node2.example.com, and I'm running mod_muc on both of them, my users may join either node to chat in the same MUC.

But if I restart a node for some reason - say general maintenance - during that node's shutdown process, mod_muc terminates all of the MUCs that belong to that node. Since all MUCs in the cluster are detected as 'belonging' to every node and this shutdown call is replicated via clustering, shutting down a single node also shuts down all of the MUCs across all remaining live nodes.

Now, if I don't give mod_muc time to shutdown properly - via `kill -9` - there's no problem. The remaining live node keeps going with it's MUCs. Obviously, anyone connected to the dead node is dropped, but all other users remain unaffected, which is what led me to look at mod_muc in the first place.

Is there any chance for a fix to this behavior? Is such a fix even possible with the way clustering is accomplished? 

Thanks for any assistance, and please let me know if this belongs somewhere else.

Ryan

**Comment by badlop** [comment]
In my test using ejabberd 15.11 (released last friday, but I think this shouldn't affect), the local rooms get not destroyed when another node is shutdown.

I have a local node, where I create a room. Another node joins the cluster, a user logged in in that node creates another room. I get this in the list of known online rooms, the first Pid is of a local process, the second is a remote process:
```
$ ejabberdctl debug
(ejabberd1@localhost)9> ets:tab2list(muc_online_room).
[{muc_online_room,{<<"room-node1">>,<<"conference.localhost">>}, <0.889.0>},
 {muc_online_room,{<<"room-node2">>,<<"conference.localhost">>}, <21632.558.0>}]
```
I shutdown the second node with:
```
$ ejabberdctl stop_kindly 5 Bye
```
Then I check again in the first node, the local room is there (and the local clients can chat in it):
```
(ejabberd1@localhost)11> ets:tab2list(muc_online_room).
[{muc_online_room,{<<"room3">>,<<"conference.localhost">>},<0.889.0>}]
```

So, can you please check the muc_online_room table contents? Maybe they give you some clue about what is happening in your cluster. Notice that a local erlang process (which handles a local room) is identified with 0.*.0, and a remote process with *.*.0.

**Comment by weiss** [comment]
It looks like this has been [fixed in ejabberd 15.07][1].

[1]: https://github.com/processone/ejabberd/commit/a16834083858b76da98b9cbfe39c5df7de9fac1a
